samples/apps/logging/logging.cpp

```diff
@@ -262,6 +262,29 @@ namespace loggingapp
         .add_query_parameter<size_t>("id")
         .install();
 
+      auto clear = [this](auto& ctx, nlohmann::json&&) {
+        auto records_handle = ctx.tx.template rw<RecordsTable>("records");
+        records_handle->foreach([&ctx](const auto& id, const auto&) {
+          update_first_write(ctx.tx, id);
+          return true;
+        });
+        records_handle->clear();
+        return ccf::make_success(true);
```

Comment:
This is rubbish, looping twice. But necessary to keep the optimisation from #2556 correct - whenever we write to a private ID (even if we don't know it uniquely), we _may_ need to update the first write table. Still calling `clear()` afterwards since we want to test its implementation e2e. Expensive but correct, which is what we want for these samples.